### PR TITLE
chore: add lint-staged and prettier

### PR DESCRIPTION
**Pull request by IronLu233**
Some code is not formatted well, like https://github.com/Magickbase/ckb-explorer-frontend/pull/9#discussion_r1245965348

And the husky in current project seems not works well, which incurs there is no check on `git commit`.

This PR make husky works well, and add `Eslint` and `Prettier` to git hooks.

### CODE REVIEW

File: .husky/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_
```
Comment (Keith-CY) [inline .husky/.gitignore:1]: What does `_` mean

File: package.json [hunk 2]
```diff
@@ -60,8 +60,9 @@
     "eslint-plugin-react-hooks": "4.6.0",
     "husky": "^7.0.1",
     "jest-styled-components": "^7.0.5",
+    "lint-staged": ">=10",
```
Comment (Keith-CY) [inline package.json:63]: Why not use the rule same as other dependencies, e.g. `"lint-staged": "^13.2.3",`

File: .husky/.gitignore (hunk 1 above)
Comment (IronLu233) [reply .husky/.gitignore:1]: husky will create a new dir named `_`, it should be ignored.
image

File: package.json (hunk 2 above)
Comment (IronLu233) [reply package.json]: It's automatically added by `npx mrm@2 lint-staged`, but seems not the best version. I've updated it.

**Review by Daryl-L — APPROVED** [review]

File: .husky/.gitignore (hunk 1 above)
Comment (Keith-CY) [reply .husky/.gitignore:1]: It could be annotated

Comment (Keith-CY) [reply .husky/.gitignore:1]: BTW, the whole directory is ignored by `_/.gitignore`

Comment (IronLu233) [reply .husky/.gitignore]: https://github.com/webpack/webpack/tree/main/.husky I retrieve the Webpack repository, `.gitignore` is not necessary. Maybe the setup script of husky and lint-staged is outdated.

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**